Example/IGListKitExamples/ViewControllers/MixedDataViewController.swift

```diff
@@ -71,13 +71,8 @@ class MixedDataViewController: UIViewController, IGListAdapterDataSource {
     //MARK: IGListAdapterDataSource
 
     func objects(for listAdapter: IGListAdapter) -> [IGListDiffable] {
-        var items = [IGListDiffable]()
-        for obj in data {
-            if selectedClass == nil || type(of: obj) == selectedClass! {
-                items.append(obj as! IGListDiffable)
-            }
-        }
-        return items
+        guard selectedClass != nil else { return data.map { $0 as! IGListDiffable } }
+        return data.filter { type(of: $0) == selectedClass! }.map { $0 as! IGListDiffable }
```

Comment:
Sure, initial filter could have both conditions. 
But it hides that if selectedClass is nil, function returns just it's input array with transformed elements.
Otherwise it filters input array and then transforms it's elements.
Version I am proposing keeps that clear.
Less condensed but more clear.